File: 백준/Gold/10836. 여왕벌/여왕벌.py
# [BOJ] 10836. 여왕벌
# 소요 시간 : 60분

import sys

size, days = map(int, sys.stdin.readline().split())
plus = [1] * (2 * size - 1)

for d in range(days):
    zero, one, two = map(int, sys.stdin.readline().split())
    for o in range(zero, zero + one):
        plus[o] += 1
    for t in range(zero + one, 2 * size - 1):
        plus[t] += 2

temp = plus[size:]
for i in range(size - 1, -1, -1):
    print(plus[i], *temp)

File: 백준/Gold/10836. 여왕벌/여왕벌.js
const input = require("fs")
  .readFileSync(
    process.platform === "linux" ? "/dev/stdin" : "../../sample.txt"
  )
  .toString()
  .trim()
  .split("\n");

const [m, n] = input[0].split(" ").map(Number);
const arr = Array(m * 2 - 1).fill(1);
for (let i = 0; i < n; i++) {
  let [zero, one, two] = input[1 + i].split(" ").map(Number);
  for (let k = zero; k < zero + one; k++) arr[k] += 1;
  for (let k = zero + one; k < zero + one + two; k++) arr[k] += 2;
}

let answer = ``;
let repeat = ``;
for (let i = m; i < arr.length; i++) {
  repeat += ` ${arr[i]}`;
}
repeat += `\n`;
for (let i = m - 1; i >= 0; i--) {
  answer += `${arr[i]}` + repeat;
}

console.log(answer);
